Realm/RLMUsernamePasswordProviderClient.mm

```diff
@@ -0,0 +1,108 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMUsernamePasswordProviderClient.h"
+#import "RLMApp_Private.hpp"
+
+@implementation RLMUsernamePasswordProviderClient
+
+static NSError* AppErrorToNSError(const realm::app::AppError& appError) {
+    return [[NSError alloc] initWithDomain:@(appError.error_code.category().name())
+                                      code:appError.error_code.value()
+                                  userInfo:@{
+                                      @(appError.error_code.category().name()) : @(appError.error_code.message().data())
+                                  }];
+}
+
+- (void)registerEmail:(NSString *)email
+             password:(NSString *)password
+    completionHandler:(RLMOptionalErrorBlock)completionHandler {
```

Comment:
@tgoyne way back in the days I recall us having a standard for { being on a new line (and it's in the C++ code guidelines). Was that never adopted or just abandoned in ObjC?
It would make this nicer to look at...

Reply:
No, we talked about it and concluded that we should stick to doing what's normal for obj-c.

Reply:
Agreed.